### ISSUE TITLE
A Margin between Histogram and discreteLegend is too narrow 

### ISSUE THREAD

**Issue by tastyminerals** [BlackEdder/ggplotd#63]
If you add a legend to histogram, the legend labels are cut. Single character labels kind of visible but still are cut, multi character labels are cut completely. Using different `discreteLegend` `width` and `height` values does not help.

![filled_hist](https://user-images.githubusercontent.com/7676160/71530284-dc2ddd00-28e0-11ea-86a4-df536eab980c.jpg)

**Comment by tastyminerals** [comment]
I quickly looked into the code and it looks like a lot of work. I could just adjust the `gg.put(Margins(15, 0, 0, 0)` to `gg.put(Margins(35, 0, 0, 0)` but it is a bad way to fix this, right? I find it weird that the longer is the string for each class the bigger margin is required and even 35 won't be enough on something like "probabilities of boxes". I thought that one can adjust the legend margins in `ggplot.d` `drawToSurface` method but can't see where we have access to the actual string there.

**Comment by BlackEdder** [comment]
Thank you for looking into it! Maybe an initial solution would be to plot the legend on top of the other plot?

Also, could you provide a working code sample, that will make it easier to have a look at.

**Comment by tastyminerals** [comment]
Here is a working code sample.
```D
import std.typecons;

void testTwoClassPlot(Tuple!(int, string)[] data, string fileName, string titleName) {
    import ggplotd.aes : aes;
    import ggplotd.geom;
    import ggplotd.ggplotd : putIn, GGPlotD, title, Margins;
    import ggplotd.legend: discreteLegend;
    import std.algorithm: map;
    auto gg = data
        .map!(a => aes!("x", "colour", "fill")(a[0], a[1], 0.45))
        .geomHist
        .putIn(GGPlotD().put(title(titleName)));
    gg.put(discreteLegend);
    gg.save(fileName~".svg");
}

void main() {
    import std.array;
    import std.range : chain, zip, repeat;
    import std.random : uniform;
    import std.range : generate, take;
    auto class1 = generate!(() => uniform(0, 20)).take(1000).array;
    auto class2 = generate!(() => uniform(0, 20)).take(1000).array;
    auto class12 = class1.chain(class2);
    auto class12Labels = "A".repeat(class1.length).chain("B".repeat(class2.length));
    auto plotData = class12.zip(class12Labels).array;
    testTwoClassPlot(plotData, "Test", "Fake data");
}
 ```

**dub.json**
```json
{
    "name": "test",
    "targetType": "executable",
    "dependencies": {
        "ggplotd": "~>1.2"
    }
}


````

**Comment by BlackEdder** [comment]
This should be fixed in 1.2.1
